File: web_ui/src/api/filterRule.ts
import http from './http'

export interface FilterRule {
  id: number
  mp_id: string  // JSON字符串，存储多个公众号ID
  mp_ids: string[]  // 解析后的公众号ID数组
  is_global: boolean  // 是否为全局规则
  rule_name: string
  remove_ids: string[]
  remove_classes: string[]
  remove_selectors: string[]
  remove_attributes: Array<{ name: string; value?: string; eq?: boolean }>
  remove_regex: string[]
  remove_normal_tag: number
  status: number
  priority: number
  created_at: string
  updated_at: string
}

export interface FilterRuleCreateParams {
  mp_id: string  // JSON字符串，存储多个公众号ID
  rule_name: string
  remove_ids?: string[]
  remove_classes?: string[]
  remove_selectors?: string[]
  remove_attributes?: Array<{ name: string; value?: string; eq?: boolean }>
  remove_regex?: string[]
  remove_normal_tag?: number
  priority?: number
}

export interface FilterRuleUpdateParams {
  rule_name?: string
  remove_ids?: string[]
  remove_classes?: string[]
  remove_selectors?: string[]
  remove_attributes?: Array<{ name: string; value?: string; eq?: boolean }>
  remove_regex?: string[]
  remove_normal_tag?: number
  status?: number
  priority?: number
}

export interface FilterRuleListResult {
  code: number
  data: {
    list: FilterRule[]
    page: {
      limit: number
      offset: number
      total: number
    }
  }
}

// 获取过滤规则列表
export const getFilterRules = (params?: { mp_id?: string; limit?: number; offset?: number }) => {
  return http.get<FilterRuleListResult>('/wx/filter-rules', { params })
}

// 获取单个过滤规则详情
export const getFilterRule = (ruleId: number) => {
  return http.get<{ code: number; data: FilterRule }>(`/wx/filter-rules/${ruleId}`)
}

// 创建过滤规则
export const createFilterRule = (data: FilterRuleCreateParams) => {
  return http.post<{ code: number; data: { id: number; message: string } }>('/wx/filter-rules', data)
}

// 更新过滤规则
export const updateFilterRule = (ruleId: number, data: FilterRuleUpdateParams) => {
  return http.put<{ code: number; data: { id: number; message: string } }>(`/wx/filter-rules/${ruleId}`, data)
}

// 删除过滤规则
export const deleteFilterRule = (ruleId: number) => {
  return http.delete<{ code: number; data: { id: number; message: string } }>(`/wx/filter-rules/${ruleId}`)
}

// 获取公众号的启用规则
export const getActiveRulesForMp = (mpId: string) => {
  return http.get<{ code: number; data: FilterRule[] }>(`/wx/filter-rules/mp/${mpId}/active`)
}

File: core/models/filter_rule.py
from .base import Base, Column, String, Integer, DateTime, Text, JSON
from datetime import datetime


class FilterRule(Base):
    """HTML过滤规则模型 - 为指定公众号配置HTML过滤规则"""
    __tablename__ = 'we_filter_rules'
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    mp_id = Column(String(255), nullable=False, index=True, comment='公众号ID')
    rule_name = Column(String(255), nullable=False, comment='规则名称')
    
    # 过滤规则配置
    remove_ids = Column(JSON, nullable=True, comment='要移除的ID列表')
    remove_classes = Column(JSON, nullable=True, comment='要移除的class列表')
    remove_selectors = Column(JSON, nullable=True, comment='CSS选择器列表')
    remove_attributes = Column(JSON, nullable=True, comment='属性过滤列表 [{"name": "attr_name", "value": "attr_value", "eq": true}]')
    remove_regex = Column(JSON, nullable=True, comment='正则表达式列表')
    remove_normal_tag = Column(Integer, default=0, comment='是否移除常见HTML元素(0-否, 1-是)')
    
    status = Column(Integer, default=1, comment='状态(1-启用, 2-禁用)')
    priority = Column(Integer, default=0, comment='优先级，数字越大优先级越高')
    
    created_at = Column(DateTime, default=datetime.now, comment='创建时间')
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now, comment='更新时间')
    
    from_attributes = True
